src/Integrations/Greenhouse/Greenhouse.php

```diff
@@ -0,0 +1,438 @@
+<?php
+
+/**
+ * Greenhouse integration class.
+ *
+ * @package EightshiftForms\Integrations\Greenhouse
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Integrations\Greenhouse;
+
+use EightshiftForms\Exception\MissingFilterInfoException;
+use EightshiftForms\Hooks\Filters;
+use EightshiftForms\Cache\Cache;
+
+/**
+ * Greenhouse integration class.
+ */
+class Greenhouse implements GreenhouseClientInterface
+{
+
+	/**
+	 * Transient name for storing GH Jobs
+	 *
+	 * @var string
+	 */
+	public const TRANSIENT_JOBS = 'ef_greenhouse_jobs';
+
+	/**
+	 * Transient LIFESPAN for storing GH Jobs. Default 6 hours like WP Rocket cache.
+	 *
+	 * @var string
+	 */
+	public const TRANSIENT_JOBS_LIFESPAN = 21600;
+
+	/**
+	 * Cache object used for caching Mailchimp responses.
+	 *
+	 * @var Cache
+	 */
+	private $cache;
+
+	/**
+	 * Constructs object
+	 *
+	 * @param Cache $transientCache Transient cache object.
+	 */
+	public function __construct(Cache $transientCache)
+	{
+		$this->cache = $transientCache;
+	}
+
+	/**
+	 * Return jobs with cache option for faster loading.
+	 *
+	 * @return array
+	 */
+	public function getJobs(): array
+	{
+		$output = $this->cache->get(static::TRANSIENT_JOBS);
+
+		if (!$output) {
+			$output = (string) wp_json_encode($this->getJobsRaw());
+
+			$this->cache->save(static::TRANSIENT_JOBS, $output, static::TRANSIENT_JOBS_LIFESPAN);
+		}
+
+		return json_decode($output, true);
+	}
+
+	/**
+	 * Return job with cache option for faster loading.
+	 *
+	 * @param string $jobId Job id to search by.
+	 *
+	 * @return array
+	 */
+	public function getJob(string $jobId): array
+	{
+		$output = array_filter(
+			$this->getJobs(),
+			function ($job) use ($jobId) {
+				if ($job['id'] === $jobId) {
+					return $job;
+				}
+			}
+		);
+
+		return (array) array_values($output)[0];
+	}
+
+	/**
+	 * API request to post job application to Greenhouse.
+	 *
+	 * @param string $jobId Job id to search.
+	 * @param array $body Field and Files to send.
+	 *
+	 * @return array
+	 */
+	public function postGreenhouseApplication(string $jobId, array $body): array
+	{
+		$this->verifyGreenhouseInfoExists();
+
+		$response = \wp_remote_post(
+			"{$this->getJobBoardUrl()}boards/{$this->getBoardToken()}/jobs/{$jobId}",
+			[
+				'headers' => $this->getHeaders(true),
+				'body' => wp_json_encode($body),
+				'data_format' => 'body',
+			]
+		);
+
+		if (is_wp_error($response)) {
+			return [];
+		}
+
+		$body = $response['body'] ?? '';
```

Comment:
I think you could use `wp_remote_retrieve_body($response)` here.

https://developer.wordpress.org/reference/functions/wp_remote_retrieve_body/

Reply:
Fixed